hazelcast/src/main/java/com/hazelcast/config/EvictionConfig.java

```diff
@@ -77,6 +85,50 @@ public EvictionConfig(int size, MaxSizePolicy maxSizePolicy, EvictionPolicy evic
         }
     }
 
+    public EvictionConfig(int size, MaxSizePolicy maxSizePolicy, String comparatorClassName) {
+        /**
+         * ===== NOTE =====
+         *
+         * Do not use setters, because they are overridden in the readonly version of this config and
+         * they cause an "UnsupportedOperationException". Just set directly if the value is valid.
+         */
+
+        // Size cannot be non-positive number
+        if (size > 0) {
+            this.size = size;
+        }
+        // Max-Size policy cannot be null
+        if (maxSizePolicy != null) {
```

Comment:
If we do all checks in `EvictionConfig` class, like `if (maxSizePolicy != null)`, with `Preconditions`, users can be informed truly in case of wrong parameter passing.